In [1]:
import pandas as pd

In [2]:
file_name = 'data_academic_performance.xlsx'

excel_file = pd.ExcelFile(file_name)

excel_file.sheet_names

['Citation',
 'Description of Categorical',
 'Description of Numerical',
 'SABER11_SABERPRO']

In [3]:
df = pd.read_excel(
    file_name,
    sheet_name='SABER11_SABERPRO'
)

df.head()

,COD_S11,GENDER,EDU_FATHER,EDU_MOTHER,OCC_FATHER,OCC_MOTHER,STRATUM,SISBEN,PEOPLE_HOUSE,Unnamed: 9,...,CC_PRO,ENG_PRO,WC_PRO,FEP_PRO,G_SC,PERCENTILE,2ND_DECILE,QUARTILE,SEL,SEL_IHE
0,SB11201210000129,F,Incomplete Professional Education,Complete technique or technology,Technical or professional level employee,Home,Stratum 4,It is not classified by the SISBEN,Three,NaN,...,71,93,79,181,180,91,5,4,2,2
1,SB11201210000137,F,Complete Secundary,Complete professional education,Entrepreneur,Independent professional,Stratum 5,It is not classified by the SISBEN,Three,NaN,...,86,98,78,201,182,92,5,4,4,4
2,SB11201210005154,M,Not sure,Not sure,Independent,Home,Stratum 2,Level 2,Five,NaN,...,18,43,22,113,113,7,1,1,1,1
3,SB11201210007504,F,Not sure,Not sure,Other occupation,Independent,Stratum 2,It is not classified by the SISBEN,Three,NaN,...,76,80,48,137,157,67,4,3,2,2
4,SB11201210007548,M,Complete professional education,Complete professional education,Executive,Home,Stratum 4,It is not classified by the SISBEN,One,NaN,...,98,100,71,189,198,98,5,4,4,2


In [4]:
df.shape

(12411, 45)

In [5]:
df.columns

Index(['COD_S11', 'GENDER', 'EDU_FATHER', 'EDU_MOTHER', 'OCC_FATHER',
       'OCC_MOTHER', 'STRATUM', 'SISBEN', 'PEOPLE_HOUSE', 'Unnamed: 9',
       'INTERNET', 'TV', 'COMPUTER', 'WASHING_MCH', 'MIC_OVEN', 'CAR', 'DVD',
       'FRESH', 'PHONE', 'MOBILE', 'REVENUE', 'JOB', 'SCHOOL_NAME',
       'SCHOOL_NAT', 'SCHOOL_TYPE', 'MAT_S11', 'CR_S11', 'CC_S11', 'BIO_S11',
       'ENG_S11', 'Cod_SPro', 'UNIVERSITY', 'ACADEMIC_PROGRAM', 'QR_PRO',
       'CR_PRO', 'CC_PRO', 'ENG_PRO', 'WC_PRO', 'FEP_PRO', 'G_SC',
       'PERCENTILE', '2ND_DECILE', 'QUARTILE', 'SEL', 'SEL_IHE'],
      dtype='object')

In [6]:
numerical_description = pd.read_excel(
    file_name,
    sheet_name='Description of Numerical'
)

numerical_description.head(20)

,Table 1. Description of numeric variables.,Unnamed: 1
0,Variable,Full name
1,MAT_S11,Mathematics
2,CR_S11,Critical Reading
3,CC_S11,Citizen Competencies S11
4,BIO_S11,Biology
5,ENG_S11,English
6,QR_PRO,Quantitative Reasoning
7,CR_PRO,Critical Reading
8,CC_PRO,Citizen Competencies SPRO
9,ENG_PRO,English


1. Does parental education influence outcomes, and if so, how much?

In [7]:
categorical_description = pd.read_excel(
    file_name,
    sheet_name='Description of Categorical'
)

categorical_description.head(30)

,Table 2. Description of categorical variables.,Unnamed: 1,Unnamed: 2,Unnamed: 3,Unnamed: 4,Unnamed: 5
0,Variable,Full Name,Levels,Variable,Full name,Levels
1,GENDER,Gender,2,DVD,DVD,2
2,EDU_FATHER,Father's education,12,FRESH,Fresh,2
3,EDU_MOTHER,Mother's education,12,PHONE,Phone,2
4,OCC_FATHER,Father's occupation,13,MOBILE,Mobile,2
5,OCC_MOTHER,Mother's occupation,13,REVENUE,Revenue,3
6,STRATUM,Stratum,7,JOB,Job,8
7,SISBEN,Sisben,6,SCHOOL_NAME,School name,3.735
8,PEOPLE_HOUSE,People in the house,13,SCHOOL_NAT,Nature of School,2
9,INTERNET,Internet,2,SCHOOL_TYPE,Type of School,4


In [8]:
df.EDU_FATHER.value_counts()

,count
EDU_FATHER,
Complete professional education,3016
Complete Secundary,2843
Complete technique or technology,1194
Incomplete Secundary,1091
Postgraduate education,1085
Complete primary,824
Incomplete primary,735
Incomplete Professional Education,425
Not sure,407


In [9]:
df.EDU_MOTHER.value_counts()

,count
EDU_MOTHER,
Complete Secundary,3106
Complete professional education,3059
Complete technique or technology,1495
Incomplete Secundary,1056
Postgraduate education,997
Complete primary,713
Incomplete primary,541
Incomplete Professional Education,502
0,388


In [12]:
father_score = df.groupby('EDU_FATHER').G_SC.mean()

father_score.sort_values(ascending=False)

,G_SC
EDU_FATHER,
Postgraduate education,176.985253
Complete professional education,167.765915
Incomplete Professional Education,166.917647
Complete technique or technology,164.033501
Incomplete technical or technological,162.101083
Not sure,160.294840
Complete Secundary,158.785790
Ninguno,156.634146
Incomplete Secundary,156.535289


In [13]:
father_score_clean = father_score.drop(
    index=['Not sure', 0],
    errors='ignore'
)

father_score_clean.sort_values(ascending=False)

,G_SC
EDU_FATHER,
Postgraduate education,176.985253
Complete professional education,167.765915
Incomplete Professional Education,166.917647
Complete technique or technology,164.033501
Incomplete technical or technological,162.101083
Complete Secundary,158.785790
Ninguno,156.634146
Incomplete Secundary,156.535289
Incomplete primary,155.729252


In [14]:
highest_father_score = father_score_clean.max()
lowest_father_score = father_score_clean.min()

difference_father = highest_father_score - lowest_father_score

print("Highest average score:", highest_father_score)
print("Lowest average score:", lowest_father_score)
print("Difference:", difference_father)

Highest average score: 176.9852534562212
Lowest average score: 155.19538834951456
Difference: 21.78986510670663


In [15]:
mother_score = df.groupby('EDU_MOTHER').G_SC.mean()

mother_score.sort_values(ascending=False)

,G_SC
EDU_MOTHER,
Postgraduate education,175.636911
Complete professional education,168.845374
Incomplete Professional Education,166.685259
Complete technique or technology,163.242140
Incomplete technical or technological,161.879765
Not sure,159.407821
Complete Secundary,158.470702
Incomplete Secundary,155.672348
Complete primary,155.336606


In [16]:
mother_score_clean = mother_score.drop(
    index=['Not sure', 0],
    errors='ignore'
)

mother_score_clean.sort_values(ascending=False)

,G_SC
EDU_MOTHER,
Postgraduate education,175.636911
Complete professional education,168.845374
Incomplete Professional Education,166.685259
Complete technique or technology,163.242140
Incomplete technical or technological,161.879765
Complete Secundary,158.470702
Incomplete Secundary,155.672348
Complete primary,155.336606
Incomplete primary,154.853974


In [17]:
highest_mother_score = mother_score_clean.max()
lowest_mother_score = mother_score_clean.min()

difference_mother = highest_mother_score - lowest_mother_score

print("Highest average score:", highest_mother_score)
print("Lowest average score:", lowest_mother_score)
print("Difference:", difference_mother)

Highest average score: 175.63691073219658
Lowest average score: 149.7058823529412
Difference: 25.931028379255395


Parental education appears to be associated with student performance. Students whose fathers had postgraduate education had an average global score of about 176.99, compared with about 155.20 for the lowest valid group, a difference of about 21.79 points. For mothers, the difference between the highest and lowest education groups was about 25.93 points. Overall, students with more highly educated parents tended to have higher global scores.

2. Which groups/stratum performs best? Are there differences between groups/stratums? Explain your analysis.

In [18]:
df.STRATUM.value_counts()

,count
STRATUM,
Stratum 3,4045
Stratum 2,4029
Stratum 1,1709
Stratum 4,1578
Stratum 5,633
Stratum 6,403
0,14


In [19]:
stratum_score = df.groupby('STRATUM').G_SC.mean()

stratum_score.sort_values(ascending=False)

,G_SC
STRATUM,
Stratum 6,181.511166
Stratum 5,177.387046
Stratum 4,172.002535
Stratum 3,163.815328
Stratum 2,158.199305
0,154.000000
Stratum 1,152.352838


In [20]:
stratum_score_clean = stratum_score.drop(
    index=[0],
    errors='ignore'
)

stratum_score_clean.sort_values(ascending=False)

,G_SC
STRATUM,
Stratum 6,181.511166
Stratum 5,177.387046
Stratum 4,172.002535
Stratum 3,163.815328
Stratum 2,158.199305
Stratum 1,152.352838


In [21]:
highest_stratum_score = stratum_score_clean.max()
lowest_stratum_score = stratum_score_clean.min()

difference_stratum = highest_stratum_score - lowest_stratum_score

print("Highest average score:", highest_stratum_score)
print("Lowest average score:", lowest_stratum_score)
print("Difference:", difference_stratum)

Highest average score: 181.51116625310175
Lowest average score: 152.3528379169105
Difference: 29.15832833619126


In [22]:
stratum_summary = df.groupby('STRATUM').G_SC.describe()

stratum_summary

,count,mean,std,min,25%,50%,75%,max
STRATUM,,,,,,,,
0,14.0,154.000000,21.321712,116.0,142.0,152.5,167.0,196.0
Stratum 1,1709.0,152.352838,21.937401,77.0,138.0,152.0,167.0,220.0
Stratum 2,4029.0,158.199305,22.278029,72.0,143.0,158.0,174.0,237.0
Stratum 3,4045.0,163.815328,21.693970,75.0,149.0,164.0,179.0,247.0
Stratum 4,1578.0,172.002535,21.469094,91.0,158.0,174.0,188.0,243.0
Stratum 5,633.0,177.387046,21.543236,37.0,165.0,179.0,192.0,246.0
Stratum 6,403.0,181.511166,21.879401,91.0,170.0,184.0,194.5,242.0


Stratum 6 performs best, with an average global score of about 181.51, while Stratum 1 has the lowest average score at about 152.35. The difference between the highest and lowest groups is about 29.16 points. The mean and median scores both increase steadily from Stratum 1 to Stratum 6, suggesting a clear relationship between stratum and academic performance. The standard deviations are similar across groups, so the main difference appears to be in the overall score level rather than variability.

3. How does each gender perform in the global assessment?

In [23]:
df.GENDER.value_counts()

,count
GENDER,
M,7368
F,5043


In [24]:
gender_score = df.groupby('GENDER').G_SC.mean()

gender_score

,G_SC
GENDER,
F,161.278207
M,163.690825


In [25]:
gender_summary = df.groupby('GENDER').G_SC.describe()

gender_summary

,count,mean,std,min,25%,50%,75%,max
GENDER,,,,,,,,
F,5043.0,161.278207,22.332939,76.0,146.0,162.0,177.0,242.0
M,7368.0,163.690825,23.582616,37.0,147.0,164.0,180.0,247.0


In [26]:
female_score = gender_score.loc['F']
male_score = gender_score.loc['M']

difference_gender = male_score - female_score

print("Female average score:", female_score)
print("Male average score:", male_score)
print("Difference:", difference_gender)

Female average score: 161.2782074162205
Male average score: 163.69082519001086
Difference: 2.4126177737903447


Male students performed slightly better in the global assessment. The average global score for males was about 163.69, while the average for females was about 161.28. The difference was about 2.41 points. The median score was also slightly higher for males, 164 compared with 162 for females. The standard deviations were similar, so the overall score distributions were fairly similar between the two groups.

4. Use your analysis to understand the results.  What did you find interesting, and can you make any recommendations or inferences?

In [27]:
print("Father education difference:", difference_father)
print("Mother education difference:", difference_mother)
print("Stratum difference:", difference_stratum)
print("Gender difference:", difference_gender)

Father education difference: 21.78986510670663
Mother education difference: 25.931028379255395
Stratum difference: 29.15832833619126
Gender difference: 2.4126177737903447


I found that socioeconomic background and parental education were more strongly associated with global scores than gender. The difference between Stratum 6 and Stratum 1 was about 29.16 points, which was the largest difference in my analysis. Parental education also showed noticeable differences, while the gender difference was only about 2.41 points. This suggests that students' social and family backgrounds may be related to academic performance. One recommendation would be to provide more academic support and educational resources to students from lower socioeconomic groups. However, these results only show associations, so we cannot conclude that these factors directly cause the differences in scores.

Additional Analysis of Student Academic Performance

In [28]:
gender_subjects = df.groupby('GENDER')[
    ['QR_PRO', 'CR_PRO', 'CC_PRO', 'ENG_PRO', 'WC_PRO', 'FEP_PRO', 'G_SC']
].mean()

gender_subjects

,QR_PRO,CR_PRO,CC_PRO,ENG_PRO,WC_PRO,FEP_PRO,G_SC
GENDER,,,,,,,
F,71.892723,61.338687,58.573865,66.532421,56.618084,145.302796,161.278207
M,81.198561,62.788409,59.606270,68.159473,51.708469,145.595548,163.690825


In [29]:
stratum_subjects = df.groupby('STRATUM')[
    ['QR_PRO', 'CR_PRO', 'CC_PRO', 'ENG_PRO', 'WC_PRO', 'FEP_PRO', 'G_SC']
].mean()

stratum_subjects

,QR_PRO,CR_PRO,CC_PRO,ENG_PRO,WC_PRO,FEP_PRO,G_SC
STRATUM,,,,,,,
0,69.785714,47.357143,46.000000,55.071429,60.500000,138.071429,154.000000
Stratum 1,70.520772,53.566413,50.918666,50.911644,49.303686,142.975424,152.352838
Stratum 2,74.616282,58.823033,55.956813,60.946389,51.149665,142.478531,158.199305
Stratum 3,78.341656,63.857849,60.150062,70.621261,53.043758,142.201236,163.815328
Stratum 4,83.215463,69.266160,66.970849,80.097592,59.592522,154.046895,172.002535
Stratum 5,86.704581,71.747235,69.230648,87.004739,63.655608,160.415482,177.387046
Stratum 6,88.362283,73.764268,71.074442,92.454094,65.585608,162.166253,181.511166


In [30]:
stratum_subjects_clean = stratum_subjects.drop(
    index=[0],
    errors='ignore'
)

stratum_subjects_clean

,QR_PRO,CR_PRO,CC_PRO,ENG_PRO,WC_PRO,FEP_PRO,G_SC
STRATUM,,,,,,,
Stratum 1,70.520772,53.566413,50.918666,50.911644,49.303686,142.975424,152.352838
Stratum 2,74.616282,58.823033,55.956813,60.946389,51.149665,142.478531,158.199305
Stratum 3,78.341656,63.857849,60.150062,70.621261,53.043758,142.201236,163.815328
Stratum 4,83.215463,69.266160,66.970849,80.097592,59.592522,154.046895,172.002535
Stratum 5,86.704581,71.747235,69.230648,87.004739,63.655608,160.415482,177.387046
Stratum 6,88.362283,73.764268,71.074442,92.454094,65.585608,162.166253,181.511166


In [31]:
high_school_scores = df[
    ['MAT_S11', 'CR_S11', 'CC_S11', 'BIO_S11', 'ENG_S11', 'G_SC']
]

high_school_scores.corr()

,MAT_S11,CR_S11,CC_S11,BIO_S11,ENG_S11,G_SC
MAT_S11,1.000000,0.609894,0.594732,0.763082,0.594903,0.643838
CR_S11,0.609894,1.000000,0.731217,0.661651,0.581045,0.653572
CC_S11,0.594732,0.731217,1.000000,0.657576,0.548531,0.634900
BIO_S11,0.763082,0.661651,0.657576,1.000000,0.593273,0.666635
ENG_S11,0.594903,0.581045,0.548531,0.593273,1.000000,0.662169
G_SC,0.643838,0.653572,0.634900,0.666635,0.662169,1.000000


In [32]:
high_school_scores.corr().G_SC.sort_values(ascending=False)

,G_SC
G_SC,1.000000
BIO_S11,0.666635
ENG_S11,0.662169
CR_S11,0.653572
MAT_S11,0.643838
CC_S11,0.634900


In [34]:
from google.colab import files

df.to_csv('academic_performance_data.csv', index=False)
files.download('academic_performance_data.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>